src/models/tabularmodel.jl

```diff
@@ -0,0 +1,79 @@
+function emb_sz_rule(n_cat)
+	min(600, round(1.6 * n_cat^0.56))
+end
+
+function _one_emb_sz(catdict, catcol::Symbol, sz_dict=nothing)
+	sz_dict = isnothing(sz_dict) ? Dict() : sz_dict
+	n_cat = length(catdict[catcol])
+	sz = catcol in keys(sz_dict) ? sz_dict[catcol] : emb_sz_rule(n_cat)
+	n_cat, sz
+end
+
+function get_emb_sz(catdict, cols, sz_dict=nothing)
+	[_one_emb_sz(catdict, catcol, sz_dict) for catcol in cols]
+end
+
+# function get_emb_sz(td::TableDataset, sz_dict=nothing)
+# 	cols = Tables.columnaccess(td.table) ? Tables.columnnames(td.table) : Tables.columnnames(Tables.rows(td.table)[1])
+# 	[_one_emb_sz(catdict, catcol, sz_dict) for catcol in cols]
+# end
+
+struct TabularModel
+	embeds
+	emb_drop
+	bn_cont
+	n_emb
+	n_cont
+	layers
+end
+
+function TabularModel(
+		layers; 
+		emb_szs,
+		n_cont,
+		out_sz,
+		ps::Union{Tuple, Vector, Number, Nothing}=nothing,
+		embed_p::Float64=0.,
+		y_range=nothing,
+		use_bn::Bool=true,
+		bn_final::Bool=false,
+		bn_cont::Bool=true,
+		act_cls=Flux.relu,
+		lin_first::Bool=true)
+
+	n_cont = Int64(n_cont)
+	if isnothing(ps)
+		ps = zeros(length(layers))
+	end
+	if ps isa Number
+		ps = fill(ps, length(layers))
+	end
```

Comment:
In addition to the comment about `isnothing(p)` being part of `linbndrop`, this can also be deleted by doing `Iterators.cycle(ps)` when building `_layers`.

Reply:
Yeah I'll try doing this.